### PR TITLE
feat: Update tests and convert to ts

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?

Refactor the test to be .ts
Rename the test repo to drop the internal reference
Remove node 4 check in tests

### CODE REVIEW

File: test/missing-node-modules.test.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+// tslint:disable-next-line:no-var-requires
+const server = require('./cli-server')(
+  process.env.SNYK_API, apiKey, notAuthorizedApiKey,
+);
+
+// ensure this is required *after* the demo server, since this will
+// configure our fake configuration too
+import * as cli from '../src/cli/commands';
+
+test('setup', (t) => {
```
Comment (dkontorovskyy) [inline test/missing-node-modules.test.ts]: make function `async`. Then you can drop `t.plan(3)` and all `then`s cos you can use `async/await`

File: test/missing-node-modules.test.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    t.pass('existing user endpoint captured');
+  });
+
+  server.listen(port, () => {
+    t.pass('started demo server');
+  });
+});
+
+test('throws when missing node_modules', async (t) => {
+  t.plan(1);
```
Comment (dkontorovskyy) [inline test/missing-node-modules.test.ts]: no need in this, as you have outer `async`

File: test/missing-node-modules.test.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  // test
+  try {
+    await cli.test(dir);
+    t.fail('should have thrown');
+  } catch (e) {
+    t.matches(e.message, /Missing node_modules folder/);
+  }
+});
+
+test('teardown', (t) => {
```
Comment (dkontorovskyy) [inline test/missing-node-modules.test.ts]: make it `async` and drop `t.plan(4)`

File: test/missing-node-modules.test.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  t.plan(4);
+
+  delete process.env.SNYK_API;
+  delete process.env.SNYK_HOST;
+  delete process.env.SNYK_PORT;
+  t.notOk(process.env.SNYK_PORT, 'fake env values cleared');
+
+  server.close(() => {
```
Comment (dkontorovskyy) [inline test/missing-node-modules.test.ts]: make it `async` and refactor https://github.com/snyk/snyk/pull/629/files#diff-1b21c43ec0490c975c740bab28eaeb3aR75 to be in `async/await` style

**Comment by lili2311** [comment]
As per chat in person will `async/await`-ify in another PR

**Pull request merged**